charts/orchestrator/templates/sonataflows.yaml

```diff
@@ -11,6 +11,22 @@
 {{- end }}
 
 {{- if .Values.sonataFlowOperator.enabled }}
+  {{- $ns := .Values.orchestrator.namespace }}
+---
+apiVersion: sonataflow.org/v1alpha08
+kind: SonataFlowClusterPlatform
+metadata:
+  name: cluster-platform
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-delete-policy": hook-fail
+    "helm.sh/hook-weight": "2" # Is created after the job to wait for the crd to be available 
```

Comment:
Shouldn't this weight should be > 2 (bigger than the weight of the SonataflowPlatform it depends on)

Reply:
Depends on when the controller will reconcile it. We can set it higher but it won't matter if the platform controller reconciles after the cluster controller.
In my opinion, the cluster controller should watch for changes to the platform CR it depends on and reconcile accordingly, so that the order in which is deployed should not matter at the end. Thoughts?

Reply:
I'm referring to this as a pure dependency issue: SonataFlowClusterPlatform must have the sonataflow-platform from its `platformRef`, otherwise, it will fail.
Maybe this isn't a real issue, if both CR are about to be created at the same time.
I referred to a problem I encountered in the past:
https://github.com/parodos-dev/orchestrator-helm-operator/issues/2
but maybe with all of the recent changes, that's no longer a concern.

Reply:
From what I can glimpse here:
https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/adf68a1a0d58a3e9a2464b62d9c6d910c166d14c/controllers/sonataflowclusterplatform_controller.go#L135-L150
The controller will watch for changes to the platform CR and reconcile the cluster platform CR accordingly.
Looking at the related issue, could this error come from a lookup in the chart that attempted to dynamically identify the platform reference CR name and use it to write it in the cluster platform CR:
```
failed to install release: no SonataFlowPlatform with the name \"sonataflow-platform\" found
```
I don't think this error message came from the `sonataflowclusterplatform` controller.